otherlibs/unix/signals.c

```diff
@@ -71,6 +71,8 @@ CAMLprim value unix_sigprocmask(value vaction, value vset)
   caml_enter_blocking_section();
   retcode = caml_sigmask_hook(how, &set, &oldset);
   caml_leave_blocking_section();
+  /* Run any handlers for just-unmasked pending signals */
+  caml_process_pending_actions();
```

Comment:
To be clear, this is another example of behaviour change from not polling in `caml_leave_blocking_section`, right?

Reply:
Yes, see the commit message.

This is related to #9802: POSIX stipulates polling points for signals in exactly two functions: `kill` and `sigprocmask` (or its `pthread_sigmask` alias). `sigprocmask` had one via `caml_leave_blocking_section`, but `kill` did not. #9802 adds a test that ensures both are present.

Reply:
Yes, I saw it.